relay-server/src/endpoints/autoscaling.rs

```diff
@@ -6,11 +6,14 @@ use std::fmt::Write;
 
 /// Returns internal metrics data for relay.
 pub async fn handle(state: ServiceState) -> (StatusCode, String) {
-    let data = match state
-        .autoscaling()
-        .send(AutoscalingMessageKind::Check)
-        .await
-    {
+    let Some(autoscaling) = state.autoscaling() else {
+        return (
+            StatusCode::SERVICE_UNAVAILABLE,
+            "Autoscaling metrics not available in Proxy mode".to_owned(),
```

Comment:
You make the assumption here you're running in proxy mode when the service isn't here, but don't actually check that.

I'd make the message a bit more generic, possibly just:

```suggestion
            "Autoscaling metrics not enabled".to_owned(),
```